File: dodian-backend/dodian-server/src/main/java/net/dodian/uber/game/model/object/RS2Object.java
package net.dodian.uber.game.model.object;

/**
 * Whoever authored this class is a massive dumbass.
 *
 * @author Logan
 * ^^^^^^^^^^^^^^^^^
 */
public class RS2Object {
    public int id, x, y, type, face, oldId;
    private java.lang.Object attachment = null;

    public RS2Object(int id, int x, int y, int type) {
        this.id = id;
        this.x = x;
        this.y = y;
        this.type = type;
    }

    public RS2Object(int id, int x, int y, int type, int face) {
        this.id = id;
        this.x = x;
        this.y = y;
        this.type = type;
        this.face = face;
    }

    public RS2Object(int id, int x, int y, int type, int face, int oldId) {
        this.id = id;
        this.x = x;
        this.y = y;
        this.type = type;
        this.face = face;
        this.oldId = oldId;
    }

    public void setAttachment(java.lang.Object o) {
        this.attachment = o;
    }

    /**
     * @return the attachment
     */
    public java.lang.Object getAttachment() {
        return attachment;
    }
}

File: game-server/bin/main/net/dodian/uber/game/model/object/RS2Object.kt
package net.dodian.uber.game.model.`object`

class RS2Object @JvmOverloads constructor(
    @JvmField var id: Int,
    @JvmField var x: Int,
    @JvmField var y: Int,
    @JvmField var type: Int,
    @JvmField var face: Int = 0,
    @JvmField var oldId: Int = 0,
) {
    private var attachment: Any? = null

    fun setAttachment(attachment: Any?) {
        this.attachment = attachment
    }

    fun getAttachment(): Any? = attachment
}
